pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -53,6 +53,16 @@ func ApplyParameters(p *v1beta1.PipelineSpec, pr *v1beta1.PipelineRun) *v1beta1.
 	return ApplyReplacements(p, stringReplacements, arrayReplacements)
 }
 
+// ApplyContexts applies the substitution from $(context.(pipelineRun|pipeline).*) with the specified values.
+// Currently supports only name substitution. Uses "" as a default if name is not specified.
+func ApplyContexts(spec *v1beta1.PipelineSpec, pipelineName string, pr *v1beta1.PipelineRun) *v1beta1.PipelineSpec {
+	stringReplacements := map[string]string{}
```

Comment:
style nit: I'd just declare these in the map initialization directly:

```
stringReplacements := map[string]string{
  "context.pipelineRun.name": pr.Name,
  "context.pipeline.name": pipelineName,
}
```

(This also lets you pass the map directly to `ApplyReplacements`, and make this method a one-liner, if you want)